### PR TITLE
[hosts] Add troubleshooting section with info on hosts detected by APM

### PR DESCRIPTION
**Pull request by colleenmcginnis**
## Description

Adds a troubleshooting section to the [Analyze and compare hosts](https://www.elastic.co/guide/en/observability/current/analyze-hosts.html) page with information on configuring APM agent to ensure that hosts detected by APM use the correct host name.

### Documentation sets edited in this PR

_Check all that apply._

- [x] Stateful (`docs/en/observability/*`)
- [x] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue

Closes https://github.com/elastic/observability-docs/issues/4145

## Checklist

- [x] Product/Engineering Review
- [ ] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [ ] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [x] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

**Review by roshan-elastic — COMMENTED** [review]
Hey @colleenmcginnis - looking good!

Could we make an addition in the troubleshooting bit to mention what the (?) means?

**Background**

- All of the hosts which have been detected by the APM agent but do not have full metrics will have the (?)
- Sometimes the host name is correct
- Sometimes it isn't and the user needs to make some config changes to APM agent (this is where the troubleshooting will be of the most value)

image

Do you think we could add a quick bit at the start of troubleshooting which defines what (?) means?

e.g. 

# Troubleshooting 

### What does 'this host has been detected by APM mean'?

When a host has been detected by [APM](https://www.elastic.co/guide/en/observability/current/apm.html) but is not collecting full metrics (e.g. via the [system integration](https://www.elastic.co/docs/current/integrations/system)), it will show as a host but with the partial metrics collected by [APM](https://www.elastic.co/guide/en/observability/current/apm.html).

### I don't recognise the host name and I see a '?' next to it

This could mean that the APM agent has not been configured to use the correct host name. Instead, the host name might be the container name or the Kubernetes pod name.

To get the correct host name, you need to set some additional configuration options, specifically system.kubernetes.node.name as described in [apm-api-kubernetes-data].

> **Note** : We would link to the "What does 'this host has been detected by APM mean'?" heading from the UI 'Learn more' as per the screenshot above.

**Review by roshan-elastic — DISMISSED** [review]
LGTM! Thanks @colleenmcginnis

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-3li00mbon-elastic-dev.vercel.app

* Built with commit c67c9ee2b4b6f0d625b4b49e83d34cc2f2a71d10

> Issues? Visit #next-docs in Slack

File: docs/en/observability/monitor-infra/analyze-hosts.asciidoc [hunk 1]
```diff
@@ -256,3 +256,45 @@ to determine if there is an outage or issue.
...
+3. How to fix it
+////
+
+[discrete]
+[[analyze-hosts-troubleshooting-detected-by-apm]]
+===  What does _this host has been detected by APM_ mean?
+
+// What the user sees/experiences (error message, UI, behavior, etc)
+In the Hosts view you might see a question mark icon (image:images/icons/questionInCircle.svg[Question mark icon])
+before a host name with a note in a tooltip stating that the host has been detected by APM.
```
Comment (mdbirnstiehl) [inline docs/en/observability/monitor-infra/analyze-hosts.asciidoc:280]:
```suggestion
In the Hosts view, you might see a question mark icon (image:images/icons/questionInCircle.svg[Question mark icon])
before a host name with a note in a tooltip stating that the host has been detected by APM.
```

File: docs/en/observability/monitor-infra/analyze-hosts.asciidoc [hunk 2]
```diff
@@ -256,3 +256,45 @@ to determine if there is an outage or issue.
...
+// Why it happens
+When a host is detected by APM, but is not collecting full metrics
+(for example, via the https://www.elastic.co/docs/current/integrations/system[system integration]),
```
Comment (mdbirnstiehl) [inline docs/en/observability/monitor-infra/analyze-hosts.asciidoc:283]:
```suggestion
(for example, through the https://www.elastic.co/docs/current/integrations/system[system integration]),
```

File: docs/en/serverless/infra-monitoring/analyze-hosts.mdx [hunk 3]
```diff
@@ -266,3 +266,38 @@ Similar to the above scenario, you may want to investigate the time period with
...
+*/}
+
+### What does _this host has been detected by APM_ mean?
+
+{/* What the user sees/experiences (error message, UI, behavior, etc) */}
+In the Hosts view you might see a question mark icon (<DocIcon type="questionInCircle" title="Question mark icon" />)
+before a host name with a note in a tooltip stating that the host has been detected by APM.
+{/* Why it happens */}
+When a host is detected by APM, but is not collecting full metrics
+(for example, via the <DocLink slug="/integrations/system">system integration</DocLink>),
```
Comment (mdbirnstiehl) [inline docs/en/serverless/infra-monitoring/analyze-hosts.mdx:288]:
```suggestion
(for example, through the <DocLink slug="/integrations/system">system integration</DocLink>),
```

File: docs/en/observability/monitor-infra/analyze-hosts.asciidoc (hunk 1 above)
Comment (mdbirnstiehl) [inline docs/en/observability/monitor-infra/analyze-hosts.asciidoc:280]:
```suggestion
In the Hosts view, you might see a question mark icon (image:images/icons/questionInCircle.svg[Question mark icon])
before a host name with a tooltip note stating that the host has been detected by APM.
```

File: docs/en/serverless/infra-monitoring/analyze-hosts.mdx [hunk 4]
```diff
@@ -266,3 +266,38 @@ Similar to the above scenario, you may want to investigate the time period with
...
+1. What the user sees/experiences (error message, UI, behavior, etc)
+2. Why it happens
+3. How to fix it
+*/}
+
+### What does _this host has been detected by APM_ mean?
+
+{/* What the user sees/experiences (error message, UI, behavior, etc) */}
+In the Hosts view you might see a question mark icon (<DocIcon type="questionInCircle" title="Question mark icon" />)
+before a host name with a note in a tooltip stating that the host has been detected by APM.
```
Comment (mdbirnstiehl) [inline docs/en/serverless/infra-monitoring/analyze-hosts.mdx:285]:
```suggestion
In the Hosts view, you might see a question mark icon (<DocIcon type="questionInCircle" title="Question mark icon" />)
before a host name with a tooltip note stating that the host has been detected by APM.
```

**Review by mdbirnstiehl — COMMENTED** [review]
Just a couple of small suggestions which you can take or leave! Otherwise, 🦭 👍

**Pull request merged**